Service/SalesReportService.php

```diff
@@ -134,146 +217,304 @@ private function convert($data)
     private function convertByProduct($data)
     {
+        $i = 0;
+        $label = array();
+        $graphData = array();
+        $backgroundColor = array();
         $products = array();
+        /* @var $entityManager EntityManager */
+        $entityManager = $this->app['orm.em'];
+        $sql = 'Select od.product_class_id From dtb_order_detail od Where od.order_detail_id = :order_detail_id';
+        $stmt = $entityManager->getConnection()->prepare($sql);
         foreach ($data as $Order) {
             /* @var $Order \Eccube\Entity\Order */
             $OrderDetails = $Order->getOrderDetails();
             foreach ($OrderDetails as $OrderDetail) {
-                /* @var $OrderDetail \Eccube\Entity\OrderDetail */
-                $ProductClass = $OrderDetail->getProductClass();
-                $id = $ProductClass->getId();
-                if (!array_key_exists($id, $products)) {
-                    $products[$id] = array(
-                        'ProductClass' => $ProductClass,
-                        'total' => 0,
-                        'quantity' => 0,
-                        'price' => 0,
-                        'time' => 0,
-                    );
+                // Get product class id
+                $params['order_detail_id'] = $OrderDetail->getId();
+                $stmt->execute($params);
+                $productClassId = $stmt->fetch(\PDO::FETCH_COLUMN);
+                if ($productClassId) {
+                    if (!array_key_exists($productClassId, $products)) {
+                        $products[$productClassId] = array(
+                            'OrderDetail' => $OrderDetail,
+                            'total' => 0,
+                            'quantity' => 0,
+                            'price' => 0,
+                            'time' => 0,
+                        );
+                    }
+                    $products[$productClassId]['quantity'] += $OrderDetail->getQuantity();
+                    $products[$productClassId]['price'] = $OrderDetail->getPriceIncTax();
+                    ++$products[$productClassId]['time'];
                 }
-                $products[$id]['total'] += $OrderDetail->getPriceIncTax();
-                $products[$id]['quantity'] += $OrderDetail->getQuantity();
-                $products[$id]['price'] = $OrderDetail->getPriceIncTax();
-                $products[$id]['time'] ++;
+            }
+        }
+        //set total
+        foreach ($products as $key => $product) {
+            $total = $product['quantity'] * $product['price'];
+            $products[$key]['total'] = $total;
+        }
+        //sort by total money
+        $products = $this->sortBy('total', $products);
+        log_info('SalesReport Plugin : product report ', array('result count' => count($products)));
+        foreach ($products as $key => $product) {
+            $total = $product['total'];
+            $backgroundColor[$i] = $this->getColor($i);
+            if ($i >= 10) {
```

Comment:
please, avoid literal. (programming basic rule)
to change `$app['config']['const']`.